sys/net/gnrc/network_layer/ipv6/nib/_nib-arsm.c

```diff
@@ -517,6 +517,9 @@ static inline uint32_t _exp_backoff_retrans_timer(uint8_t ns_sent,
     uint32_t tmp = random_uint32_range(NDP_MIN_RANDOM_FACTOR,
                                        NDP_MAX_RANDOM_FACTOR);
 
+    /* this is a given since ns_sent is only incremented in _probe_nbr when
+     * it is less than (NDP_MAX_NS_NUMOF + 2) */
+    assert(ns_sent < 32);
     /* backoff according to  https://tools.ietf.org/html/rfc7048 with
      * BACKOFF_MULTIPLE == 2 */
     tmp = ((1 << ns_sent) * retrans_timer * tmp) / US_PER_MS;
```

Comment:
I could be wrong, but this sounds like a separate issue not directly related to this PR.

Reply:
> For 16bit/8bit platforms, should not this `1` also be a `uint32_t` or even `uint64_t` as `NDP_MAX_NS_NUMOF` can be `17` ?

Huh, I don't get your statement... `tmp` is an `uint32_t`. 17 + 2 is on all platforms < 32.

Reply:
Or do you mean, the `1` should actually be `1UL`?

Reply:
(if it is the latter I agree with @bergzand: this is a matter of a separate PR)

Reply:
@bergzand it is not related to silencing `scan-build` but has been revealed by running the static analyzer. And unfortunately we do not have `llvm`support with 8/16bit platforms.
If it is silenced, it would be forgotten.

@miri64 yes I meant the `1` should be converted to an `uint32`, not sure of the size of `1UL` in our 8 and 16bit platform and in general `((1 << ns_sent) * retrans_timer * tmp)` should not overflow in its type.

Reply:
@cladmi things can be fixed in a separate PR *while* this PR is open ;-) and thus not be forgotten. Since I now understand your remark better, I will do that as soon as I have a keyboard under my fingers.

Reply:
@cladmi see https://github.com/RIOT-OS/RIOT/pull/10369